Sources/Target/POSIX/ELFProcess.cpp

```diff
@@ -298,26 +311,15 @@ ErrorCode ELFProcess::updateInfo() {
     }
 
 #if defined(OS_LINUX)
-    // For some reason, on Android 5.1 and up some ELF segments get loaded in
-    // memory multiple smaller contiguous regions instead of being loaded as a
-    // single big mapping.
-    // This means we can't properly identify the beginning of the segment by
-    // just looking at the start address of the mapping; we have to inspect
-    // contiguous mappings instead and see if they are contiguous in the input
-    // file and then take the start address of the first mapping of the segment
-    // we're inspecting.
-    do {
-      prevMri = mri;
-      error = getMemoryRegionInfo(mri.start - 1, mri);
-      if (error != kSuccess) {
-        goto mri_error;
-      }
-    } while (mri.backingFile == prevMri.backingFile &&
-             mri.backingFileInode == prevMri.backingFileInode &&
-             mri.protection == prevMri.protection &&
-             mri.backingFileOffset + mri.length == prevMri.backingFileOffset);
-
-    _loadBase = prevMri.start;
+    // Here, we use the interpreter's base address
+    // instead of the binary's. The interpreter
+    // has an ELF header just the same as the
+    // binary, so we can get most of the necessary
+    // info from it.
+    // If this fails for any reason, an easy fallback
+    // is to grab the PHDR and subtract the size of
+    // the ELF header. This should be good enough, though.
+    _loadBase = getAuxiliaryVectorValue(AT_BASE);
```

Comment:
We don't need _loadBase at all. We can use the PHDR to find the ELF header of the binary (the same way the modified function does things) and then start extracting info that way. We can also forego the ELF header validation, since if auxv exists, then we have an ELF. to begin with.